core/modules/field/field.module

```diff
@@ -562,8 +562,21 @@ function field_config_delete_validate(Config $active_config, $all_changes) {
 function field_config_create(Config $staging_config) {
   $config_name = $staging_config->getName();
   if (strpos($config_name, 'field.field.') === 0) {
+    $field_data_table_exists = FALSE;
+
     $field = $staging_config->get();
-    field_create_field($field);
+    $schema = _field_sql_storage_schema($field);
```

Comment:
This makes me a little nervous because it's introducing hard-coupling between field_sql_storage and field module. We don't support other field storages however, so this might be acceptable. We already introduced something similar in `field_validate_field()`, where we also explicitly check if the DB tables exist.

I think handling this we should try either of the following:

1. Determine why `field_config_create()` is getting called twice and avoid doing it in the first place.
2. Create a follow-up issue to introduce `field_storage_exists()` and then put a TODO both here and in `field_validate_field()`.

Reply:
> Determine why field_config_create() is getting called twice and avoid doing it in the first place.

It isn't called twice as far as I have noticed. Did you notice that happening?
> Create a follow-up issue to introduce field_storage_exists() and then put a TODO both here and in field_validate_field().

Sounds good. We could use `_field_info_collate_fields()` for this as it provides the field storage details.

Reply:
> It isn't called twice as far as I have noticed. Did you notice that happening?

This may be an incorrect assumption on my part. I would have thought that it getting called twice would be the reason for why we need to check if the table exists, because it would have already been called once before and created them. If that's not the case, why do these tables already exist at the time this is called?

Reply:
Ahh, I should have realised. The reason is here: https://github.com/backdrop/backdrop-issues/issues/3224#issuecomment-903198426